### ISSUE TITLE
Distributed Index primary key attribute

### ISSUE THREAD

**Issue by njspok** [yiisoft/yii2-sphinx#45]
For distributed index attribute primaryKey not assigned. In my application make this problem.

```php
// yiisoft/yii2-sphinx/Schema.php
// Schema::findColumns()
    protected function findColumns($index)
    {
        $sql = 'DESCRIBE ' . $this->quoteSimpleIndexName($index->name);
        try {
            $columns = $this->db->createCommand($sql)->queryAll();
        } catch (\Exception $e) {
            $previous = $e->getPrevious();
            if ($previous instanceof \PDOException && strpos($previous->getMessage(), 'SQLSTATE[42S02') !== false) {
                // index does not exist
                // https://dev.mysql.com/doc/refman/5.5/en/error-messages-server.html#error_er_bad_table_error
                return false;
            }
            throw $e;
        }

        if (empty($columns[0]['Agent'])) {
            foreach ($columns as $info) {
                $column = $this->loadColumnSchema($info);
                $index->columns[$column->name] = $column;
                if ($column->isPrimaryKey) {
                    $index->primaryKey = $column->name; // local index assign pk attribute
                }
            }
        } else {
            // Distributed index :
            $agent = $this->getIndexSchema($columns[0]['Agent']);
            $index->columns = $agent->columns;

            // FIX distributed index not assigned pk
            foreach ($index->columns as $name => $column) {
                if ($column->isPrimaryKey) {
                    $index->primaryKey = $column->name;
                }
            }
        }

        return true;
    }
```

**Comment by KhristenkoYura** [comment]
Wait  [pull request](https://github.com/yiisoft/yii2-sphinx/pull/40)

**Comment by klimov-paul** [comment]
Resolved by commit 65cc2eb11ea54720225b4e7d8503e19cae61aa96
